src/document/json/rga_tree_split.ts

```diff
@@ -541,7 +541,7 @@ export class RGATreeSplit<T extends RGATreeSplitValue> {
     const isRemote = !!latestCreatedAtMapByActor;
     const changes: Array<Change> = [];
     const createdAtMapByActor = new Map();
-    const nodesToDelete = [];
+    const nodesToDelete: Array<RGATreeSplitNode<T>> = [];
```

Comment:
I thought the type of the list was automatically inferred so I didn't have to specify the type. Are there any rules for specifying the type?

Reply:
All pushed values come from "candidates", all of which are `Array<RGATreeSplitNode<T>>`.

There seems to be no situation where the type can change within that context...!

https://github.com/yorkie-team/yorkie-js-sdk/blob/2beaee9325ebb68392f10825b4263d0179156c32/src/document/json/rga_tree_split.ts#L536-L558

Reply:
Thanks for the explanation. I tested how [type inference](https://www.typescriptlang.org/docs/handbook/type-inference.html) works in TypeScript as follows.

```typescript
let x = 3; // infferred by `number`
x = 'a';   // Error: Type `string` is not assignable to type `number`

const array = ['1']; // inffered by Array<number>;
array.push(2);       // Error: Argument of type 'number' is not assignable to parameter of type 'string'

const array2 = [];   // inffered by Array<any>;
array2.push(1);
array2.push('str');
```

When initializing to an empty array, it becomes an array of `any`, so I found that it is correct to declare while specifying the type when initializing the array.